src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -4286,6 +4286,76 @@ module Queries = struct
         Mina_lib.runtime_config mina
         |> Runtime_config.to_yojson |> Yojson.Safe.to_basic )
 
+  let fork_config =
+    let rec map_results ~f = function
+      | [] ->
+          Result.return []
+      | r :: rs ->
+          let open Result.Let_syntax in
+          let%bind r' = f r in
+          let%map rs = map_results ~f rs in
+          r' :: rs
+    in
+    field "fork_config"
+      ~doc:
+        "The runtime configuration for a blockchain fork intended to be a \
+         continuation of the current one."
+      ~typ:(non_null Types.json)
+      ~args:Arg.[]
+      ~resolve:(fun { ctx = mina; _ } () ->
+        match Mina_lib.best_tip mina with
+        | `Bootstrapping ->
+            `Assoc [ ("error", `String "Daemon is bootstrapping") ]
+        | `Active best_tip -> (
+            let block = Transition_frontier.Breadcrumb.(block best_tip) in
+            let global_slot =
+              Mina_block.blockchain_length block |> Unsigned.UInt32.to_int
+            in
+            let accounts_or_error =
+              Transition_frontier.Breadcrumb.staged_ledger best_tip
+              |> Staged_ledger.ledger
+              |> Ledger.foldi ~init:[] ~f:(fun _ accum act -> act :: accum)
+              |> map_results
+                   ~f:Runtime_config.Json_layout.Accounts.Single.of_account
+            in
+            let protocol_state =
+              Transition_frontier.Breadcrumb.protocol_state best_tip
+            in
+            match accounts_or_error with
+            | Error e ->
+                `Assoc [ ("error", `String e) ]
+            | Ok accounts ->
+                let runtime_config = Mina_lib.runtime_config mina in
+                let ledger = Option.value_exn runtime_config.ledger in
+                let previous_length =
+                  let open Option.Let_syntax in
+                  let%bind proof = runtime_config.proof in
+                  let%map fork = proof.fork in
+                  fork.previous_length + global_slot
+                in
+                let fork =
+                  Runtime_config.Fork_config.
+                    { previous_state_hash =
+                        State_hash.to_base58_check
+                          protocol_state.previous_state_hash
+                    ; previous_length =
+                        Option.value ~default:global_slot previous_length
+                    ; previous_global_slot = global_slot
+                    }
+                in
+                let update =
+                  Runtime_config.make
+                    ~ledger:
+                      { ledger with
+                        base = Accounts accounts
+                      ; add_genesis_winner = Some false
```

Comment:
What exactly do you mean by "won't work"? I managed to feed this config into a sandbox node and it started poroducing blocks. On the other hand, when I left this field unchanged, the node was trying to add the genesis winner account twice (once because it was there in the genesis ledger in the config and the other because of this option). Should I be removing that account from the genesis ledger then?

Reply:
Ah, of course. Can you add a comment to this effect?